.travis.yml

```diff
@@ -285,6 +288,23 @@ install:
   - if [ "${BUILD_DATALAD_EXTENSION:-}"x = "datalad-container"x ]; then travis_retry sudo eatmydata apt-get install singularity-container; fi
 
 script:
+  - if [ ! -z "${_DL_BENCHMARK:-}" ]; then
+        pip install asv;
+        asv machine --yes ;
+        pip install -e .;
+        sed -i -e 's,"branches,//branches,g' -e 's,"pythons,//pythons,g' -e 's,.asv/,travisasv/,g' asv.conf.json ;
+        git tag '__bench_target__';
+        git rev-parse HEAD __bench_target__;
+        asv run -E existing --set-commit-hash $(git rev-parse __bench_target__);
+        git reset --hard;
+        git checkout origin/master;
+        git rev-parse HEAD __bench_target__;
+        pip install -e .;
+        sed -i -e 's,"branches,//branches,g' -e 's,"pythons,//pythons,g' -e 's,.asv/,travisasv/,g' asv.conf.json ;
+        asv run -E existing --set-commit-hash $(git rev-parse origin/master);
+        asv compare origin/master __bench_target__;
```

Comment:
There should be some env variable which would show the merge target, or if Travis tests merged state, then I guess it would be HEAD^2 or something like that? That would allow to benchmark 0.11.x PRs as well

Reply:
Also may be this whole code could go into a script, which could then get used even locally or across cis?

Reply:
re variables: dunno about those. But it would be great to test against whatever the base is, indeed.

re into-a-script: I didn't do that, because that script would be in a branch, and possibly change between `HEAD` and base. Of course this could all be mitigated by explicitly checking things out (`cat-file ...`) from `HEAD`, but I had no such ambitions for now.

Reply:
> There should be some env variable which would show the merge target, or if Travis tests merged state, then I guess it would be HEAD^2 or something like that? That would allow to benchmark 0.11.x PRs as well

GitHub's 'refs/pull/*/merge' ref is the result of the PR merged into the mainline, so HEAD^1 would be either origin/master or origin/0.11.x.   Though I'd find defining the target branch an environment variable cleaner because then the same logic would work for non-PR branches.

Even if we don't bother making this work for 0.11.x in this PR, I think we should skip the benchmarks if the run is for the master branch where `__bench_target__` would be master.

Reply:
> re variables: dunno about those.

I _think_ the suggestion is to have a variable in .travis.yml (say `_DL_MERGE_TARGET`) that points to "origin/master" in `master:.travis.yml` and to "origin/0.11.x" in `0.11.x:.travis.yml`.  (It could also default to "origin/master" when not defined.)

Reply:
Re branch variable from https://docs.travis-ci.com/user/environment-variables/#convenience-variables

TRAVIS_BRANCH:
- for push builds, or builds not triggered by a pull request, this is the name of the branch.
- for builds triggered by a pull request this is the name of the branch targeted by the pull request.
- for builds triggered by a tag, this is the same as the name of the tag (TRAVIS_TAG).

Reply:
Which also reminds that this run is relevant only for PRs and should exit cleanly in regular branch build

Reply:
Regarding the script - I don't see how anything is different from just coding it in .travis.yml which could also differ between branches. But it will be easier to check/sync across branches if it was in a script

Reply:
> Re branch variable from https://docs.travis-ci.com/user/environment-variables/#convenience-variables

The variables you list, along with TRAVIS_PULL_REQUEST, are fine for detection.  But as I said, "I'd find defining the target [merge] branch as an environment variable cleaner because then the same logic would work for non-PR branches."

> Which also reminds that this run is relevant only for PRs and should exit cleanly in regular branch build

AFAICT this is the [same issue I mentioned above](https://github.com/datalad/datalad/pull/3544#discussion_r304982395), but you propose to exclude all non-PR runs rather than just run for  master or 0.11.x.

Reply:
It seems there is much that I am/was not aware of. I'll leave it here for now, and try to come back to it later. In the meantime I can satisfy my desires by plopping this temporarily onto relevant PRs.

Reply:
>  I'll leave it here for now, and try to come back to it later. 

I can take it over.

> In the meantime I can satisfy my desires by plopping this temporarily onto relevant PRs.

Beware that the current form isn't doing the intended comparison AFAICT.  As I was reworking this, I realized that `asv run` reports to be benchmarking the same commit (compare [here][0] to [here][1]).  The recipe in this PR expects that HEAD is benchmarked if "branches" isn't configured, but, based on the comment in asv.conf.json and local testing, it looks like it benchmarks master if no branches are specified.

[0]: https://travis-ci.org/datalad/datalad/jobs/560509048#L1116
[1]: https://travis-ci.org/datalad/datalad/jobs/560509048#L1205